### PR TITLE
Compare all egress fields in data plane reconciler for triggering updates

### PR DESCRIPTION
**Pull request by pierDipi**
We are missing some recently added fields to compare.

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java [hunk 1]
```diff
@@ -240,8 +240,12 @@ private boolean egressEquals(DataPlaneContract.Egress e1, DataPlaneContract.Egre
     return Objects.equals(e1.getUid(), e2.getUid())
       && Objects.equals(e1.getConsumerGroup(), e2.getConsumerGroup())
       && Objects.equals(e1.getDestination(), e2.getDestination())
+      && Objects.equals(e1.hasDiscardReply(), e2.hasDiscardReply())
       && Objects.equals(e1.getReplyUrl(), e2.getReplyUrl())
       && Objects.equals(e1.getReplyToOriginalTopic(), e2.getReplyToOriginalTopic())
+      && Objects.equals(e1.getDeliveryOrder(), e2.getDeliveryOrder())
+      && Objects.equals(e1.getVReplicas(), e2.getVReplicas())
+      && Objects.equals(e1.getFeatureFlags(), e2.getFeatureFlags())
```
Comment (aavarghese) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:248]: Should we compare `Reference` field too?

Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:248]: it's not needed, for now, it is immutable and the fact that there is a version (resourceVersion) in the field might trigger unexpected side effects.

Comment (aavarghese) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:248]: ok

**Review by aavarghese — APPROVED** [review]
Tests in-progress...

/lgtm

**Pull request merged**